gn3/api/wgcna.py

```diff
@@ -0,0 +1,25 @@
+"""endpoint to run wgcna analysis"""
+from flask import Blueprint
+from flask import request
+from flask import current_app
+from flask import jsonify
+
+from gn3.computations.wgcna import call_wgcna_script
+
+wgcna = Blueprint("wgcna", __name__)
+
+
+@wgcna.route("/run_wgcna", methods=["POST"])
+def run_wgcna():
+    """run wgcna:output should be a json with a the data"""
+
+    wgcna_data = request.json
+
+    wgcna_script = current_app.config["WGCNA_RSCRIPT"]
+
+    results = call_wgcna_script(wgcna_script, wgcna_data)
```

Comment:
**issue:** It's a bit confusing that you pass in wgcna_data that is a first order function, instead of the data itself .i.e. you pass in `request.json` instead of `request.json()`. Prefer the former.

Reply:
Cool. I guess you mean request.get_json() maybe we could use that for clarity but request.json is a property  that delegates to request.get_json() so they kinda  work the same. Maybe the confusion arises coz of the request.json() method in the requests python library.